### PR TITLE
Added conditional on change handler to dynamic stepper

### LINKED ISSUE #23: Should be able to close active step
#### What is the expected behavior?
On the Dynamic Stepper example, you should be able to close the active step if it has a value selected. 
Clicking the header should collapse the step.

#### What is the current behavior?
Clicking the header of the open step does nothing.
click scroll header

#### How to recreate?
1. Clone the repo & cd into the folder.
2. Run `yarn && yarn start`
3. Navigate to http://localhost:3000/dynamic-stepper
4. Select one of the options for step one (the step will close automatically)
5. Re-open step 1 by clicking on the step header
6. Click the step header again and observe the step does not close.

#### Additional Information

This is happening because the click handler of the header is currently just setting the active step to the index of the selected step. We need to add conditional logic to see if that is already the selected step, and if so, collapse it (set active step to undefined/-1).

### PR DESCRIPTION
**Pull request by bkarambe**
Fixes [#23](https://github.com/etn-ccis/blui-react-design-patterns/issues/23)  BLUI-3712

#### Changes proposed in this Pull Request:
- Added a conditional on click handler in dynamic stepper

#### Screenshots:
https://user-images.githubusercontent.com/25982779/209767812-2a2a515b-4c3e-4dbf-add1-07d6b4fadb77.mov

#### To Test:
- yarn start
- Navigate to Dynamic Stepper
- Add step and chose an option
- Click on Step title to open
- Click on the expanded step title to close

### CODE REVIEW

File: src/pages/dynamic-stepper/index.tsx [hunk 1]
```diff
@@ -174,9 +174,15 @@ export const DynamicStepper = (): JSX.Element => {
...
-                                        }}
+                                        onClick={
+                                            index === activeStep
+                                                ? (): void => {
+                                                      setActiveStep(-1);
+                                                  }
+                                                : (): void => {
+                                                      setActiveStep(index);
+                                                  }
+                                        }
```
Comment (joebochill) [inline src/pages/dynamic-stepper/index.tsx:185]: You can simplify this a little bit:
```suggestion
                                        onClick={ (): void => {
                                              setActiveStep(index === activeStep ? -1 : index);
                                        } }
```

**Review by huayunh — COMMENTED** [review]
Sometimes a user might see a blue text for adding a new step, although the row itself is still correctly disabled. Can you make sure the text is also picking up the disabled style? Change otherwise looks nice.

https://user-images.githubusercontent.com/8997218/210076954-4c9e4294-2622-4de6-b1a6-5db8ad7cdf0a.mov

**Comment by daileytj** [comment]
@huayunh the issue you just mentioned is occurring in `dev` and should be logged as a separate issue to avoid scope creep on this story.

**Comment by huayunh** [comment]
This is a one-liner style fix on [line 236](https://github.com/etn-ccis/blui-react-design-patterns/blob/bugfix/3712-23-active-step-close-stepper/src/pages/dynamic-stepper/index.tsx#L236) instead of a functional fix. When you have 5 hours left on the story, I don't think patching this would be a big scope creep like with some other PRs.

**Comment by daileytj** [comment]
> This is a one-liner style fix on [line 236](https://github.com/etn-ccis/blui-react-design-patterns/blob/bugfix/3712-23-active-step-close-stepper/src/pages/dynamic-stepper/index.tsx#L236) instead of a functional fix. When you have 5 hours left on the story, I don't think patching this would be a big scope creep like with some other PRs.

I had not looked at the functionality behind it but makes sense. I just added the fix so it's done.

**Review by huayunh — APPROVED** [review]

**Pull request merged**